dsgrid/dataset/growth_rates.py

```diff
@@ -34,24 +42,58 @@ def apply_growth_rate_123(
     pyspark.sql.DataFrame
 
     """
-    orig_columns = initial_value_df.columns
-    initial_value_df, base_year = _check_model_years(
-        dataset, initial_value_df, growth_rate_df, model_year_column
+
+    initial_value_df, growth_rate_df = _process_exponential_growth_rate(
+        dataset,
+        initial_value_df,
+        growth_rate_df,
+        model_year_column,
+        pivoted_columns,
+    )
+
+    df = apply_annual_multiplier(
+        initial_value_df,
+        growth_rate_df,
+        time_columns,
+        pivoted_columns,
     )
 
+    return df
+
+
+def apply_annual_multiplier(
+    initial_value_df,
+    growth_rate_df,
+    time_columns,
+    pivoted_columns,
+):
+    """Applies annual growth rate to the initial_value dataframe as follows:
+    P(t) = P0 * r(t)
+    where:
+      P(t): quantity at year t
+      P0: initial quantity
+      r(t): growth rate per year t (relative to P0)
+
+    Parameters
+    ----------
+    initial_value_df : pyspark.sql.DataFrame
```

Comment:
We are not at all consistent with these docstrings. If the functions are only going to be called internally and we want to specify the type only once, I would declare it as a type annotation in the function definition instead. No need to change the code again; this is just what I’ve been doing. We can try to standardize this.